pyproject.toml

```diff
@@ -0,0 +1,75 @@
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+  "setuptools>=42",
+  "setuptools_scm",
+  "wheel",
+]
+
+[project]
+name = "eofs"
+description = "EOF analysis in Python"
+license = {text = "GPL-3.0"}
+authors = [
+    {name = "Andrew Dawson", email = "dawson@atm.ox.ac.uk"},
+]
+requires-python = ">=3.8"
+classifiers = [
+  "Programming Language :: Python :: 3 :: Only",
+  "Programming Language :: Python :: 3.8",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+]
+dynamic = [
+  "version",
+]
+dependencies = [
+  "numpy",
+]
+[project.optional-dependencies]
+extras = [
+  "cdms2",
+  "cdutil",
+  "iris",
+  "xarray",
+]
+[project.urls]
+documentation = "https://ajdawson.github.io/eofs"
+homepage = "https://github.com/ajdawson/eofs"
+repository = "https://github.com/ajdawson/eofs"
+
+[tool.setuptools]
+license-files = ["COPYING"]
+include-package-data = true
+[tool.setuptools.packages.find]
+where = ["lib"]
+[tool.setuptools.package-data]
+eofs_examples = [
+  "example_data/*",
+]
+eofs_tests = ["data/*"]
+[tool.setuptools.dynamic]
+readme = {file = "README.md", content-type = "text/markdown"}
+
+[tool.setuptools_scm]
```

Comment:
Using setuptools_scm instead of versioneer b/c that one is better maintained and "blessed" by the Python packaging community.